phpunit/block-supports/elements-test.php

```diff
@@ -0,0 +1,305 @@
+<?php
+
+/**
+ * Test the elements block support.
+ *
+ * @package Gutenberg
+ */
+
+class WP_Block_Supports_Elements_Test extends WP_UnitTestCase {
+	/**
+	 * @var string|null
+	 */
+	private $test_block_name;
+
+	public function set_up() {
+		parent::set_up();
+		$this->test_block_name = null;
+	}
+
+	public function tear_down() {
+		WP_Style_Engine_CSS_Rules_Store_Gutenberg::remove_all_stores();
+		unregister_block_type( $this->test_block_name );
+		$this->test_block_name = null;
+		parent::tear_down();
+	}
+
+	/**
+	 * Registers a test block type with the provided color block supports.
+	 *
+	 * @param array $color_settings The color block support settings used for elements.
+	 */
+	public function register_block_with_color_settings( $color_settings ) {
+		$this->test_block_name = 'test/element-block-supports';
+
+		register_block_type(
+			$this->test_block_name,
+			array(
+				'api_version' => 3,
+				'attributes'  => array(
+					'style' => array(
+						'type' => 'object',
+					),
+				),
+				'supports'    => array(
+					'color' => $color_settings,
+				),
+			)
+		);
+	}
+
+	/**
+	 * Tests that elements block support applies the correct classname.
+	 *
+	 * @covers ::gutenberg_render_elements_support
+	 *
+	 * @dataProvider data_elements_block_support_class
+	 *
+	 * @param array  $color_settings  The color block support settings used for elements support.
+	 * @param array  $elements_styles The elements styles within the block attributes.
+	 * @param string $block_markup    Original block markup.
+	 * @param string $expected_markup Resulting markup after application of elements block support.
+	 */
+	public function test_elements_block_support_class( $color_settings, $elements_styles, $block_markup, $expected_markup ) {
+		$this->register_block_with_color_settings( $color_settings );
+
+		$block = array(
+			'blockName' => $this->test_block_name,
+			'attrs'     => array(
+				'style' => array(
+					'elements' => $elements_styles,
+				),
+			),
+		);
+
+		$actual = gutenberg_render_elements_support( $block_markup, $block );
+
+		$this->assertMatchesRegularExpression(
+			$expected_markup,
+			$actual,
+			'Position block wrapper markup should be correct'
```

Comment:
```suggestion
			'Elements block wrapper markup should be correct'
```

Reply:
Thanks, case-sensitive search and replace got me 🙂